src/components/camera/Client.ts

```diff
@@ -0,0 +1,79 @@
+import { type Camera, MimeType, Properties } from './Camera'
+import { CameraServiceClient } from '../../gen/component/camera/v1/camera_pb_service.esm'
+import type Client from '../../Client'
+import type { HttpBody } from '../../gen/google/api/httpbody_pb'
+import pb from '../../gen/component/camera/v1/camera_pb.esm'
+import { promisify } from '../../utils'
+
+export class CameraClient implements Camera {
+  private client: Client | undefined
```

Comment:
Wanted to mention that JS has native private class members nowadays if we want to enforce these rules in both JS and TS: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields